### PR TITLE
📊 energy: Update irena capacity data

### PR DESCRIPTION
**Pull request by pabloarosado**
* Update all capacity data from IRENA. All data comes now from a different file, which is much more complete, and includes on- and off-grid capacity, and even non-renewable sources (although we are currently ignoring them for the existing charts).
  * NOTE: `photovoltaic_cost_and_capacity` steps are duplicated from the previous version, so there is no need to review them.
* Archive old steps.

### CODE REVIEW

File: etl/steps/data/meadow/irena/2024-11-01/renewable_capacity_statistics.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+]
+
+
+def sanity_check_inputs(tables: Dict[str, Table]) -> None:
+    # Sanity checks.
+    error = "Sheet names have changed."
+    assert set(tables) == set(EXPECTED_SHEET_NAMES), error
+
+    error = "Columns have changed in the 'All Data' sheet."
+    assert set(EXPECTED_COLUMNS) == set(tables["All Data"].columns), error
```
Comment (spoonerf) [inline etl/steps/data/meadow/irena/2024-11-01/renewable_capacity_statistics.py:45]: Great examples of how to use asserts better, really helpful for me to see, thanks!

**Review by spoonerf — APPROVED** [review]
LGTM! Is it expected that there's no chart diff? I see some charts have changed so I was a little surprised by that.

File: etl/steps/data/garden/energy/2024-11-01/photovoltaic_cost_and_capacity.meta.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        title: Solar photovoltaic module price
+        presentation:
+          title_public: Solar photovoltaic module price
+      cost_source:
+        title: Source for each value of cost data
+        unit: ''
+      cumulative_capacity:
+        title: Solar photovoltaic cumulative capacity
+        presentation:
+          # NOTE: A title_public is needed, since a display.name is defined.
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply etl/steps/data/garden/energy/2024-11-01/photovoltaic_cost_and_capacity.meta.yml]: Good point. It's a bit obscure in this case. The display name is propagated from one of the combined datasets (column `photovoltaics` from the Farmer & Lafond dataset). I've improved that comment to clarify it.

File: etl/steps/data/meadow/irena/2024-11-01/renewable_capacity_statistics.py (hunk 1 above)
Comment (pabloarosado) [reply etl/steps/data/meadow/irena/2024-11-01/renewable_capacity_statistics.py:45]: Thank you! :)

**Comment by pabloarosado** [comment]
> LGTM! Is it expected that there's no chart diff? I see some charts have changed so I was a little surprised by that.

Thanks! Argh, it seems something went wrong on the staging server, I had already approved all charts. I'll check what's going on.

**Pull request merged**